low-level/solution/lr2/storage/manifest.py

```diff
@@ -0,0 +1,421 @@
+                "version": psu.get("hardware-version", "NA"),
+                "part_number": psu.get("part-number", "NA"),
+                "last_updated": int(time.time()),
+                "specifics": [{
+                    "location": psu.get("location", "NA"),
+                    "dc12v": psu.get("dc12v", "NA"),
+                    "dc5v": psu.get("dc5v", "NA"),
+                    "dc33v": psu.get("dc33v", "NA"),
+                    "dc12i": psu.get("dc12i", "NA"),
+                    "dc5i": psu.get("dc5i", "NA"),
+                    "model": psu.get("model", "NA"),
+                    "status": psu.get("status", "NA"),
+                    "mfg_date": psu.get("mfg-date", "NA")
+                }]
+            }
+            data.append(psu_dict)
+            logger.debug(self.log.svc_log(
+                f"PSU Manifest Data:{data}"))
+        return data
+
+    def get_drives_info(self):
+        """Update and return drives information in specific format."""
+        data = []
+        drives = ENCL.get_realstor_encl_data("drives")
+        for drive in drives:
+            slot = drive.get("slot", -1)
+            if slot == -1:
+                continue
+            drive_dict = {
+                "uid": drive.get("durable-id", "NA"),
+                "type": drive.get("type", "NA"),
+                "description": drive.get("description", "NA"),
+                "product": drive.get("object-name", "NA"),
+                "manufacturer": drive.get("vendor", "NA"),
+                "serial_number": drive.get("serial-number", "NA"),
+                "version": drive.get("hardware-version", "NA"),
+                "part_number": drive.get("part-number", "NA"),
+                "last_updated": int(time.time()),
+                "specifics": [{
+                    "drive-serial-number": drive.get("serial-number")[:8],
+                    "model": drive.get("model", "NA"),
+                    "slot": slot,
+                    "architecture": drive.get("architecture", "NA"),
+                    "interface": drive.get("interface", "NA"),
+                    "usage": drive.get("usage", "NA"),
+                    "current_job_completion": drive.get("current-job-completion", "NA"),
+                    "speed": drive.get("speed", "NA"),
+                    "size": drive.get("size", "NA"),
+                    "enclosure_wwn": drive.get("enclosure-wwn", "NA"),
+                    "status": drive.get("status", "NA"),
+                    "ssd_life_left": drive.get("ssd-life-left", "NA"),
+                    "led_status": drive.get("led-status", "NA"),
+                    "temperature": drive.get("temperature", "NA"),
+                    "location": drive.get("location", "NA")
+                }]
+            }
+            data.append(drive_dict)
+            logger.debug(self.log.svc_log(
+                f"Drive Manifest Data:{data}"))
+        return data
+
+    @staticmethod
+    def get_fan_specfics(fan):
```

Comment:
Please correct the typo `get_fan_specfics` -> `get_fan_specifics`

Reply:
Done